packages/valory/skills/decision_maker_abci/states/decision_receive.py

```diff
@@ -59,6 +59,12 @@ def end_block(self) -> Optional[Tuple[SynchronizedData, Enum]]:
             return None
 
         synced_data, event = cast(Tuple[SynchronizedData, Enum], res)
+        decision_receive_timestamp = self.most_voted_payload_values[-1]
+
+        synced_data = cast(
+            SynchronizedData,
+            synced_data.update(decision_receive_timestamp=decision_receive_timestamp),
+        )
```

Comment:
Just checking: do we want to update the timestamp in all cases? Even if mech response failed?

Reply:
This is now only updated in the case of a DONE event